### PR TITLE
Extract Locale to a separate file and add tests

### PR DESCRIPTION
**Pull request by naivekook**
### What
- Extracted `Locale` object to a separate file 
- Added unit tests for it.
- Changed behavior of `new` constructor, now it returns default locale if `cc` is empty string

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ Cargo.lock
 
 # These are backup files generated by rustfmt
 **/*.rs.bk
+
+# IntelliJ project files
+.idea
```
Comment (naivekook) [inline .gitignore:13]: ignore all inside the local IntelliJ folder

File: src/locale.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+impl Locale {
+    /// Returns a new Locale object with the given country code and optional language code.
+    /// If country code is empty returns default locale instead.
+    pub fn new(cc: &str, lc: Option<&str>) -> Self {
+        let cc = String::from(cc);
+        if cc.is_empty() {
+            Locale::default()
+        } else {
```
Comment (naivekook) [inline src/locale.rs:47]: new condition for empty `cc`
@jjdo the standard 639-1 is quite strict, should we add additional checks to be sure that `cc` is real `cc`? For example to check if the string is right two symbol length and contains only characters?

File: src/locale.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    type Err = Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        debug_assert!(!s.is_empty());
+        let mut split = s.split('-');
+        let cc = split.next();
+        let mut lc = split.next();
+        if lc.is_some() && lc.unwrap().is_empty() {
+            lc = None
+        }
```
Comment (naivekook) [inline src/locale.rs:83]: @jjdo I think rust has a more optimal way to do this :)

File: src/locale.rs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+impl FromStr for Locale {
+    type Err = Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        debug_assert!(!s.is_empty());
+        let mut split = s.split('-');
+        let cc = split.next();
+        let mut lc = split.next();
+        if lc.is_some() && lc.unwrap().is_empty() {
```
Comment (naivekook) [inline src/locale.rs]: @jjdo I think rust has a more optimal way to do this :)

File: src/locale.rs (hunk 2 above)
Comment (jjdo) [reply src/locale.rs:47]: Are you suggesting to validate these value on every new Locale object ?
I thing the validation would make perfect sense when we implement the write endpoints but for the read ones, I am not so sure, IMHO this should be the caller's responsibility. The docstring clearly states which values are valid for the two parameters.  
In principle, passing an invalid value would result only in a 400 Not found error or perhaps a connection error when passed in the domain name. Isn't it ?

File: src/locale.rs (hunk 4 above)
Comment (jjdo) [reply src/locale.rs]: Well spotted:

`let lc = split.next().filter(|t|!t.is_empty());`

File: src/output.rs [hunk 5]
```diff
@@ -148,6 +72,7 @@ impl Output {
 #[cfg(test)]
 mod tests_locale {
```
Comment (jjdo) [inline src/output.rs]: Please move the Locale unit tests the locale.rs file.

File: src/locale.rs (hunk 2 above)
Comment (naivekook) [reply src/locale.rs:47]: You have a point, right now it looks like overkill. So I leave it as is right now and add a to-do inside our TODO.md 👍

**Comment by jjdo** [comment]
👍

File: src/locale.rs [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+impl Display for Locale {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        if self.lc.is_some() {
+            write!(f, "{}-{}", self.cc, self.lc.as_ref().unwrap())
+        } else {
+            write!(f, "{}", self.cc)
+        }
+    }
+}
```
Comment (jjdo) [inline src/locale.rs:71]: IMHO this one should be done with a pattern. Perhaps
```
match self.lc {
    Some(ref locale) => write!(f, "{}-{}", self.cc, locale),
    None =>  write!(f, "{}", self.cc)
}
```

Comment (naivekook) [reply src/locale.rs:71]: thanks for a hint!

**Comment by naivekook** [comment]
PR is ready to merge, I don't have permissions to do this

**Comment by teolemon** [comment]
@naivekook fixed 👍

**Pull request merged**